homeassistant/components/unifi/controller.py

```diff
@@ -191,37 +198,63 @@ def event_update(self):
             return False
 
-        for unifi_config in hass.data[UNIFI_CONFIG]:
+        unifi_config = {}
+        for config in hass.data[UNIFI_CONFIG]:
             if (
-                self.host == unifi_config[CONF_HOST]
-                and self.site_name == unifi_config[CONF_SITE_ID]
+                self.host == config[CONF_HOST]
+                and self.site_name == config[CONF_SITE_ID]
             ):
-                self.unifi_config = unifi_config
+                unifi_config = config
                 break
 
-        options = dict(self.config_entry.options)
-
-        if CONF_BLOCK_CLIENT in self.unifi_config:
-            options[CONF_BLOCK_CLIENT] = self.unifi_config[CONF_BLOCK_CLIENT]
-
-        if CONF_TRACK_CLIENTS in self.unifi_config:
-            options[CONF_TRACK_CLIENTS] = self.unifi_config[CONF_TRACK_CLIENTS]
+        old_options = dict(self.config_entry.options)
+        new_options = {}
 
-        if CONF_TRACK_DEVICES in self.unifi_config:
-            options[CONF_TRACK_DEVICES] = self.unifi_config[CONF_TRACK_DEVICES]
+        if (
+            CONF_BLOCK_CLIENT in unifi_config
+            and self.option_block_clients != unifi_config[CONF_BLOCK_CLIENT]
+        ):
+            new_options[CONF_BLOCK_CLIENT] = unifi_config[CONF_BLOCK_CLIENT]
 
-        if CONF_TRACK_WIRED_CLIENTS in self.unifi_config:
-            options[CONF_TRACK_WIRED_CLIENTS] = self.unifi_config[
-                CONF_TRACK_WIRED_CLIENTS
-            ]
+        if (
+            CONF_TRACK_CLIENTS in unifi_config
+            and self.option_track_clients != unifi_config[CONF_TRACK_CLIENTS]
+        ):
+            new_options[CONF_TRACK_CLIENTS] = unifi_config[CONF_TRACK_CLIENTS]
 
-        if CONF_DETECTION_TIME in self.unifi_config:
-            options[CONF_DETECTION_TIME] = self.unifi_config[CONF_DETECTION_TIME]
-
-        if CONF_SSID_FILTER in self.unifi_config:
-            options[CONF_SSID_FILTER] = self.unifi_config[CONF_SSID_FILTER]
-
-        hass.config_entries.async_update_entry(self.config_entry, options=options)
+            if (
+                CONF_TRACK_WIRED_CLIENTS in unifi_config
+                and self.option_track_wired_clients
+                != unifi_config[CONF_TRACK_WIRED_CLIENTS]
+            ):
+                new_options[CONF_TRACK_WIRED_CLIENTS] = unifi_config[
+                    CONF_TRACK_WIRED_CLIENTS
+                ]
+
+        if (
+            CONF_TRACK_DEVICES in unifi_config
+            and self.option_track_devices != unifi_config[CONF_TRACK_DEVICES]
+        ):
+            new_options[CONF_TRACK_DEVICES] = unifi_config[CONF_TRACK_DEVICES]
+
+        if (
+            CONF_DETECTION_TIME in unifi_config
+            and self.option_detection_time
+            != timedelta(seconds=unifi_config[CONF_DETECTION_TIME])
+        ):
+            new_options[CONF_DETECTION_TIME] = unifi_config[CONF_DETECTION_TIME]
+
+        if (
```

Comment:
Can we generalize this? Something like this

https://github.com/home-assistant/home-assistant/blob/691e3f61415c1e74b93e7ceaa27d5ea4b7d06de3/homeassistant/helpers/entity_registry.py#L255-L262

Reply:
If you extract the updates checking into a standalone method, it will be easier to test too.